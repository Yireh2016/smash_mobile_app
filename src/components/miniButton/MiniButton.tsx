import * as React from 'react';
import { StyleSheet, View } from 'react-native';
import theme from '../../theme/theme';
import MiniArrow from '../../assets/MiniArrow';

import { P2 } from '../texts/Texts';
import { TouchableOpacity } from 'react-native-gesture-handler';

interface MiniButtonProps {
  label: string;
  onPress: () => vooid;
}

const styles = StyleSheet.create({
  layout: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
  },
  miniArrowContainer: {
    marginLeft: 6,
  },
});

const MiniButton: React.FunctionComponent<MiniButtonProps> = ({
  label,
  onPress,
}) => {
  return (
    <TouchableOpacity testID="MiniButton" onPress={onPress}>
      <View style={styles.layout}>
        <P2>{label}</P2>
        <View style={styles.miniArrowContainer}>
          <MiniArrow color={theme.palette.secondary} />
        </View>
      </View>
    </TouchableOpacity>
  );
};

export default MiniButton;
