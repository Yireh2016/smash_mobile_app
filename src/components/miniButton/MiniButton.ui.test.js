import * as React from 'react';
import renderer from 'react-test-renderer';
import MiniButton from './MiniButton';

it('MiniButton renders correctly', () => {
  const MiniButtonProps = {
    onPress: () => jest.mock(),
    label: 'Test',
  };
  const Element = () => {
    return <MiniButton {...MiniButtonProps} />;
  };

  const redered = renderer.create(<Element />).toJSON();
  expect(redered).toMatchSnapshot();
});
