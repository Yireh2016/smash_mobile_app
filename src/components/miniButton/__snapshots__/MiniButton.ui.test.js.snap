// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MiniButton renders correctly 1`] = `
<RNGestureHandlerButton
  collapsable={false}
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  rippleColor={0}
  testID="MiniButton"
>
  <View
    accessible={true}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontFamily": "OpenSans-Regular",
            "fontSize": 14,
          }
        }
      >
        Test
      </Text>
      <View
        style={
          Object {
            "marginLeft": 6,
          }
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight={11.368}
          bbWidth={12.184}
          focusable={false}
          height={11.368}
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "flex": 0,
                "height": 11,
                "width": 12,
              },
            ]
          }
          vbHeight={11.368}
          vbWidth={12.184}
          width={12.184}
          xml="


<svg
  xmlns=\\"http://www.w3.org/2000/svg\\"
  viewBox=\\"0 0 12.184 11.368\\"
>
  <path
    d=\\"M4.685,3.647l1-1a.772.772,0,0,1,1,0l5,5a.772.772,0,0,1,0,1l-5,5a.772.772,0,0,1-1,0h-1a.76.76,0,0,1,0-1l3-3h-7a1.451,1.451,0,0,1-1-1v-1c0-.347.653,0,1,0h7l-3-3A.754.754,0,0,1,4.685,3.647Z\\"
    transform=\\"translate(0.315 -2.464)\\"
    fill=\\"#29b782\\"
  />
</svg>

"
          xmlns="http://www.w3.org/2000/svg"
        >
          <RNSVGGroup>
            <RNSVGPath
              d="M4.685,3.647l1-1a.772.772,0,0,1,1,0l5,5a.772.772,0,0,1,0,1l-5,5a.772.772,0,0,1-1,0h-1a.76.76,0,0,1,0-1l3-3h-7a1.451,1.451,0,0,1-1-1v-1c0-.347.653,0,1,0h7l-3-3A.754.754,0,0,1,4.685,3.647Z"
              fill={4280924034}
              matrix={
                Array [
                  1,
                  0,
                  0,
                  1,
                  0.315,
                  -2.464,
                ]
              }
              propList={
                Array [
                  "fill",
                ]
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
  </View>
</RNGestureHandlerButton>
`;
